### PR TITLE
Prepare for circom v2.2.0 upgrade

### PR DESCRIPTION
**Pull request by Chengxuan**
first part of https://github.com/hyperledger-labs/zeto/issues/100
- picks up https://github.com/iden3/snarkjs/releases/tag/v0.7.5 which contains https://github.com/iden3/snarkjs/pull/518 that removed the extra check field
- Move syntax to use [anonymous components](https://docs.circom.io/circom-language/anonymous-components-and-tuples/)
- ~picks up [circom v2.2.0](https://github.com/iden3/circom/releases/tag/v2.2.0) Maintained the usage of `--O2` optimization since we are using groth16~ (going to be the next PR as changes turns out to be more complicated)

### CODE REVIEW

File: zkp/js/test/nf_anon_nullifier.js [hunk 1]
```diff
@@ -194,9 +194,9 @@ describe("main circuit tests for Zeto non-fungible tokens with anonymity using n
       err = e;
     }
     // console.log(err);
-    expect(err).to.match(/Error in template Zeto_319 line: 76/);
+    expect(err).to.match(/Error in template Zeto_319 line: 55/);
     expect(err).to.match(
-      /Error in template CheckHashesForTokenIdAndUri_88 line: 58/,
+      /Error in template CheckHashesForTokenIdAndUri_88 line: 51/,
```
Comment (EnriqueL8) [inline zkp/js/test/nf_anon_nullifier.js:199]: did it odd that we patten match on line number...

File: zkp/circuits/nf_anon.circom [hunk 2]
```diff
@@ -40,30 +40,17 @@ template Zeto(nInputs, nOutputs) {
...
+    inputOwnerPublicKeys[i]= [inputOwnerPubKeyAx, inputOwnerPubKeyAy];
   }
 
-  component checkInputHashes = CheckHashesForTokenIdAndUri(nInputs);
-  checkInputHashes.tokenIds <== tokenIds;
-  checkInputHashes.tokenUris <== tokenUris;
-  checkInputHashes.commitments <== inputCommitments;
-  checkInputHashes.salts <== inputSalts;
-  checkInputHashes.ownerPublicKeys <== inputOwnerPublicKeys;
+  CheckHashesForTokenIdAndUri(nInputs)(tokenIds <== tokenIds, tokenUris <== tokenUris, commitments <== inputCommitments, salts <== inputSalts, ownerPublicKeys <== inputOwnerPublicKeys);
```
Comment (EnriqueL8) [inline zkp/circuits/nf_anon.circom:51]: trying to understand why this change of format everywhere is necessary

**Review by EnriqueL8 — COMMENTED** [review]
Looks good - a few comment just around formatting

**Comment by Chengxuan** [comment]
@EnriqueL8 Thanks for the comments.  for Both the formatting and the reason for switching them to use anonymous component syntax are to prepare for organize those loose attributes into "buses" (a concept similar to `struct`s in golang). The hope is those attributes will naturally form buses and we can do a search and replace for all the occurrence. <-- that's why I kept them in a single line.

**Review by jimthematrix — APPROVED** [review]
LGTM. approved with a minor request

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ name: release
 on:
   push:
     tags:
-      - 'v*.*.*'
+      - "v*.*.*"
```
[the remark this answers is not in the document]
Comment (Chengxuan) [reply .github/workflows/release.yaml:7]: vscode settings are currently ignored. will remove it from gitignore

File: zkp/js/test/nf_anon_nullifier.js (hunk 1 above)
Comment (EnriqueL8) [reply zkp/js/test/nf_anon_nullifier.js:199]: @Chengxuan thoughts on this?

Comment (Chengxuan) [reply zkp/js/test/nf_anon_nullifier.js:199]: this is by far the most reliable way to determine the error. Maybe able to find some alternative along the way.
